### PR TITLE
Refactor/switch config format to yaml

### PR DESCRIPTION
**Pull request by taomoh**
### JIRA link ###

N/A

### Change description ###

Switched config file format from json to yaml. Also took this opportunity to add property for civil-service API base URL and 
to change port number from default template 3100 to 3001

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+security:
+  referrerPolicy: 'origin'
+port: 3001
+timeout: 30000
+services:
+  civilService:
+    url: 'http://localhost:8765'
+    microservice: 'civil_service'
+  idam:
+    authorizationURL: 'https://idam-web-public.aat.platform.hmcts.net/login'
```
Comment (mounikahmcts) [inline config/default.yaml:10]: To be consistent  we should point to localhost idam urls here as well. In custom environment variables we can have them pointed to respective variables via variables.

File: README.md [hunk 2]
```diff
@@ -152,7 +150,7 @@ Make sure you have those values set correctly for your application.
 
 ### Healthcheck
 
-The application exposes a health endpoint (https://localhost:3100/health), created with the use of
+The application exposes a health endpoint (https://localhost:3001/health), created with the use of
```
Comment (LeonardoPalmeiro) [inline README.md:153]: why we have to change the port?

**Review by mounikahmcts — APPROVED** [review]

File: config/custom-environment-variables.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+services:
```
Comment (LeonardoPalmeiro) [inline config/custom-environment-variables.yaml:1]: 1) There is no   "useCSRFProtection": "true" 
2) are we mixing refactor with IDAM task?

File: config/default.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+security:
```
Comment (LeonardoPalmeiro) [inline config/default.yaml:1]: there is no   "useCSRFProtection": true,

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (taomoh) [reply README.md:153]: 3100 is the default port provided by the expressjs template

File: config/custom-environment-variables.yaml (hunk 3 above)
Comment (taomoh) [reply config/custom-environment-variables.yaml:1]: There is no mix with IDAM task. The properties will be used by @osheaf01 and @laarmada once merged into master

**Review by LeonardoPalmeiro — APPROVED** [review]
perfect

**Pull request merged**